File: client/src/main/java/ru/kidesoft/ticketplace/client/domain/presenter/SceneManager.java
package ru.kidesoft.ticketplace.client.domain.presenter;

import ru.kidesoft.ticketplace.client.domain.executor.ThrowableLambda;

import java.util.function.Function;

public interface SceneManager {
    void openScene(ControllerType type) throws Exception;

    <T> void setUserData(ThrowableLambda.ThrowableSupplier<T> runnable) throws Exception;
}

File: adapter/src/main/kotlin/ru/kidesoft/ticketplace/adapter/ui/presenter/SceneManager.kt
package ru.kidesoft.ticketplace.adapter.ui.presenter

import ru.kidesoft.ticketplace.adapter.ui.presenter.ui.Alert
import ru.kidesoft.ticketplace.adapter.ui.presenter.ui.Notification
import ru.kidesoft.ticketplace.adapter.ui.presenter.ui.Scene

interface SceneManager : Notification, Alert {
    fun openScene(scene : Scene)
}

interface ViewManager {
    fun getSceneManager() : SceneManager
}